### DISCUSSION TITLE
Empty cluster list

### DISCUSSION THREAD

**Discussion by luckask3** [rackslab/Slurm-web#322] [Q&A] [answered]
Hello everyone,

After install and configuration Slurm and Slurm-web I have a problem when I try access slurm-web in my browser.
The following message is displayed:

"Empty cluster list
Try to refresh...."

But my controller node and nodes are working correctly. Does anyone know the reason for this error?

PARTITION AVAIL  TIMELIMIT  NODES  STATE NODELIST
debug*       up   infinite      3   idle BRDG3F[04-05,07]

My slurm-wlm version is:
slurm-wlm 23.11.4

I'm also confused by this munge error:

-0400 Info:      Unauthorized credential for client UID=0 GID=0

**Comment by rezib** [comment]
Hello @luckask3, do you see error messages in Slurm-web gateway and agent services?

There are some hints to check services logs, depending on your configuration, in the troubleshooting guide: https://docs.rackslab.io/slurm-web/misc/troubleshooting.html

An extract of the gateway logs (optionally anonymized) would be helpful in this case to diagnose this issue.

**Reply by luckask3** [reply]
I think everything is fine with the gateway and agent service.

In the gateway configuration in the [agent] url section I didn't change it, it's the default. Should I fill in something in this parameter?

[agents]

# List of Slurm-web agents URL
#
# This parameter is required.
url=
https://cluster1/agent
https://cluster2/agent

Log Gateway:
slurm-web-gateway.service - Slurm-web HPC dashboard gateway
     Loaded: loaded (/usr/lib/systemd/system/slurm-web-gateway.service; enabled; preset: enabled)
     Active: active (running) since Fri 2024-07-26 15:42:50 -04; 2 days ago
   Main PID: 19342 (slurm-web-gatew)
      Tasks: 1 (limit: 77121)
     Memory: 38.5M (peak: 40.6M)
        CPU: 30.854s
     CGroup: /system.slice/slurm-web-gateway.service
             └─19342 /usr/bin/python3 /usr/libexec/slurm-web/slurm-web-gateway

Jul 29 09:32:27  slurm-web-gateway[19342]: Thread-128 (process_request_thread): [INFO] Retrieving info from agent at url https://cluster1/agent
Jul 29 09:32:27  slurm-web-gateway[19342]: Thread-128 (process_request_thread): [ERROR] Unable to retrieve agent info from url https://cluster1/agent: [ConnectionError] HTTPSConnec>
Jul 29 09:32:27  slurm-web-gateway[19342]: Thread-128 (process_request_thread): [INFO] Retrieving info from agent at url https://cluster2/agent
Jul 29 09:32:27  slurm-web-gateway[19342]: Thread-128 (process_request_thread): [ERROR] Unable to retrieve agent info from url https://cluster2/agent: [ConnectionError] HTTPSConnec


Log Agent:
slurm-web-agent.service - Slurm-web HPC dashboard agent
     Loaded: loaded (/usr/lib/systemd/system/slurm-web-agent.service; enabled; preset: enabled)
     Active: active (running) since Fri 2024-07-26 13:38:49 -04; 2 days ago
   Main PID: 18742 (slurm-web-agent)
      Tasks: 1 (limit: 77121)
     Memory: 38.1M (peak: 38.1M)
        CPU: 29.585s
     CGroup: /system.slice/slurm-web-agent.service
             └─18742 /usr/bin/python3 /usr/libexec/slurm-web/slurm-web-agent

Jul 26 13:38:49 systemd[1]: Started slurm-web-agent.service - Slurm-web HPC dashboard agent.
Jul 26 13:38:49 slurm-web-agent[18742]: MainThread: [WARNING] Caching is disabled
Jul 26 13:38:49 slurm-web-agent[18742]: MainThread: [INFO] Running slurm-web-agent application
Jul 26 13:38:49 slurm-web-agent[18742]:  * Serving Flask app 'slurm-web-agent'
Jul 26 13:38:49 slurm-web-agent[18742]:  * Debug mode: off
Jul 26 13:38:49 slurm-web-agent[18742]: MainThread: [INFO] WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.

**Reply by rezib** [reply]
The default `/etc/slurm-web/gateway.ini` is empty in the package so yes, you might have changed it in some way :smile: 

It looks like you copy/pasted the [full example configuration file](https://docs.rackslab.io/slurm-web/conf/conf/gateway.html#example) from the documentation? I should probably state this more clearly in the documentation, but you should not do this. The example is present to illustrate how to define parameters, it MUST NOT be used as a basis from your custom configuration.

You have to define `[agents]`>`url` parameter according to your setup, as [explained in the quickstart guide](https://docs.rackslab.io/slurm-web/install/quickstart.html#_initial_setup). This will fix the errors `[ERROR] Unable to retrieve agent info from url https://cluster1/agent: [ConnectionError] HTTPSConnec>` in gateway service logs.

**Reply by luckask3** [reply]
Okay, so each URL address should be the node in my cluster and not the controller, right?

I had only changed the [service] interface= tag with the address of my controller and the [ui] pointing to the same host that is my controller as well, the other parameters I did not fill in.

**Reply by rezib** [reply, marked as answer]
> Okay, so each URL address should be the node in my cluster and not the controller, right?

`[agents]`>`urls` are the URL to your Slurm-web agents. If you have one cluster, you have one agent. If it is running on the same node as the gateway as a native service, you should probably set `url=http://localhost:5012`.

**Reply by luckask3** [reply]
Perfect, thanks for your help, I added the whole network range as 0.0.0.0 and it worked.
